tools/codegen/api/types.py

```diff
@@ -240,6 +242,47 @@ class DispatcherArgument:
     def __str__(self) -> str:
         return f"{self.type} {self.name}"
 
+@dataclass(frozen=True)
+class DispatcherSignature:
```

Comment:
Can you add a class comment describing what the dispatcher signature is for and how it differs from NativeSignature / CppSignature ? Side note: maybe rename CppSignature to CppApiSigntature or CppFrontendSignature